### PR TITLE
security-context -> enclave

### PR DESCRIPTION
**Pull request by mikaelarguedas**
Signed-off-by: Mikael Arguedas <mikael.arguedas@gmail.com>

### CODE REVIEW

**Comment by mikaelarguedas** [comment]
CI at https://github.com/ros2/sros2/pull/198#issuecomment-612017072

File: rcl/CMakeLists.txt [hunk 1]
```diff
@@ -58,7 +58,7 @@ set(${PROJECT_NAME}_sources
   src/rcl/subscription.c
   src/rcl/time.c
   src/rcl/timer.c
-  src/rcl/validate_security_context_name.c
+  src/rcl/validate_enclave_name.c
```
Comment (ivanpauno) [inline rcl/CMakeLists.txt:61]: I still like including explicitly `security` in the name, though I don't mind strongly.

File: rcl/include/rcl/arguments.h [hunk 2]
```diff
@@ -42,7 +42,8 @@ typedef struct rcl_arguments_t
 #define RCL_PARAM_FILE_FLAG "--params-file"
 #define RCL_REMAP_FLAG "--remap"
 #define RCL_SHORT_REMAP_FLAG "-r"
-#define RCL_SECURITY_CONTEXT_FLAG "--security-context"
+#define RCL_ENCLAVE_FLAG "--enclave"
```
Comment (ivanpauno) [inline rcl/include/rcl/arguments.h:45]: Same here, especially in the long version of the flag.

Comment (mikaelarguedas) [reply rcl/include/rcl/arguments.h:45]: this set of PRs is applying the renaming as described in the final state of the [design doc](https://github.com/ros2/design/pull/274).
Please comment there for suggesting changes to flag names to keep discussion in a single place

File: rcl/test/rcl/test_security.cpp [hunk 3]
```diff
@@ -120,16 +120,16 @@ class TestGetSecureRoot : public ::testing::Test
...
 
   /* Security directory is set, but there's no matching directory */
-  /// Wrong security context
+  /// Wrong enclave
   EXPECT_EQ(
-    rcl_get_secure_root("some_other_security_context", &allocator),
+    rcl_get_secure_root("some_other_enclave", &allocator),
     (char *) NULL);
   rcl_reset_error();
 }
```
Comment (ruffsl) [inline rcl/test/rcl/test_security.cpp:135]: @mikaelarguedas , I was cross checking that this PR had also updated the recent changes from https://github.com/ros2/rcl/pull/609 , but I'm not seeing the added `successScenarios_local_root_security_context` test in this PR.

Comment (mikaelarguedas) [reply rcl/test/rcl/test_security.cpp:135]: Good point :+1:

The test was part of this branch as it had been rebased but was not updated.
The test did fail on the rebased job https://ci.ros2.org/job/ci_windows/9974/testReport/junit/rcl/TestGetSecureRoot/successScenarios_local_root_security_context/ and should be fixed by https://github.com/ros2/rcl/pull/612/commits/f0d86fc9cfd65257686f55e62b94c1706e0c3eec

**Pull request merged**